### ISSUE TITLE
Combine Image Metadata

### ISSUE THREAD

**Issue by mattmotoki** [HACC2018/ohia.ai#14]
Create a single file with a row for each of the following plants:

* [Native Plants](https://drive.google.com/file/d/1lIs8638gBxmWgDSTc84gnGSSHgMsyYXs/view)
* [Invasive Plants](http://www.oahuisc.org/target-pests/)
* Common Plants: hibiscus, plumeria, pineapple, ginger, anthurium, heliconia, bougainvillea

Add a column for each of the following fields:
* Hawaiian Name
* Common Name
* Genus
* Species
* Status (Endangered, Native, Endemic, Invasive)
* Story
* Uses

Upload all data to our shared [data directory](https://drive.google.com/drive/u/1/folders/1lgRqxc8dWflkXn8a2dRB8GBswCe_bZpZ).

**Comment by mattmotoki** [comment]
Tasks:
* download/scrape images of plants
* obtain labels: genera for classification, bounding box labels for detection/localization
* filter out all non-plant images

All the plants on this list: https://wildlifeofhawaii.com/flowers/category/native-status/native-plants/

**Comment by mattmotoki** [comment]
Create a single file with a row for each of the following plants:

* [Native Plants](https://drive.google.com/file/d/1lIs8638gBxmWgDSTc84gnGSSHgMsyYXs/view)
* [Invasive Plants](http://www.oahuisc.org/target-pests/)
* Common Plants
   - hibiscus
   - plumeria
   - pineapple
   - ginger
   - anthurium
   - heliconia
   - bougainvillea

Add a column for each of the following fields:
* Hawaiian Name
* Common Name
* Genus
* Species
* Status (Endangered, Native, Endemic, Invasive)
* Story
* Uses

**Comment by mattmotoki** [comment]
Combined the Native, Invasive, Common plants into a single file called plant_categories.csv  The file is in the data directory. 

All fields except for the "plant_name" (the category that we are predicting) need to filled in.

**Comment by windybasket** [comment]
Input plant_categories_v2 with most fields filled in.

**Comment by mattmotoki** [comment]
Manually removed extra information; e.g., pineapple nutritional information.  Cleaned up text with some regular expressions
```
clean_text <- function( x){
  x <- gsub('\n', ' ', x)         # replace new lines with space
  x <- gsub('\\[\\d+\\]', '', x)  # remove references [\\d+]
  x <- gsub('\\"+', '\\"', x)     # remove double quotes
  x <- gsub('\\s+', ' ', x)       # remove extra whitespace
  x <- trimws(x)                  # remove leading and trailing ws
  x
}
```

For example, 
```
The ovaries develop into berries, which coalesce into a large, compact, multiple fruit. The fruit of a pineapple is arranged in two interlocking helices, eight in one direction, 13 in the other, each being a Fibonacci number.[12]

The pineapple carries out CAM photosynthesis,[13] fixing carbon dioxide at night and storing it as the acid malate, then releasing it during the day aiding photosynthesis.

In the wild, pineapples are pollinated primarily by hummingbirds.[2][14] Certain wild pineapples are foraged and pollinated at night by bats.[15]
```
gets converted to 
```
The ovaries develop into berries, which coalesce into a large, compact, multiple fruit. The fruit of a pineapple is arranged in two interlocking helices, eight in one direction, 13 in the other, each being a Fibonacci number. The pineapple carries out CAM photosynthesis, fixing carbon dioxide at night and storing it as the acid malate, then releasing it during the day aiding photosynthesis. In the wild, pineapples are pollinated primarily by hummingbirds. Certain wild pineapples are foraged and pollinated at night by bats. 
```
New data in `plant_categories_v3.csv`

**Comment by mattmotoki** [comment]
Added Sora's "canoe plants" to the list.  These plants are pretty well-known in Hawaii and add about 500 images per plant to our dataset.  The updated data is in `local_meta_v4.csv`.

I changed the name of the files because I am working with metadata from multiple sources.  `plant_categores` and `plant_meta` was getting confusing.  `local` will refer to the set of plants that will be predicted in our app.  

Each plant in `local_meta_v4.csv` has a corresponding sub-folder in our shared images drive.
